src/pages/homepage/Homepage.jsx

```diff
@@ -11,17 +11,42 @@ export default function Homepage() {
   const [pageNumber, setPageNumber] = useState(1);
   const [limitNumber, setLimitNumber] = useState(5);
   const navigate = useNavigate();
-
   const { topic } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sortByQuery = searchParams.get("sort_by");
+  const orderByQuery = searchParams.get("order_by");
+  const location = useLocation();
+
```

Comment:
Please add [a loading pattern](https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns) to this page, it was very unclear if nothing was working or if the API request was working when I loaded the page ⌛